New item tiles

bhollis opened the pull request:
This is a very rough work in progress branch for a second stab at new item tiles. The majority of it is in `DarkTile.tsx` which is a bad name for it.

<img width="247" alt="screen shot 2018-10-08 at 6 37 35 pm" src="https://user-images.githubusercontent.com/313208/46641939-45350380-cb29-11e8-97e6-e57eb0f6468f.png">

@SunburnedGoose if you wanna take a crack at this feel free!

SunburnedGoose commented:
The issue I've been working on here is the Masterwork graphics. I need to drop the grey border and replace it with the gold border/graphic of the MW icon.  When I do that, the grey box below it looks out of place.

bhollis commented:
I’ve been thinking that maybe we want to edit the masterwork graphics or stick with our own indicators. I don’t think the in-game ones look very good.

bhollis commented:
Before:
<img width="1402" alt="screen shot 2018-11-11 at 12 46 28 am" src="https://user-images.githubusercontent.com/313208/48310892-5a97b600-e54b-11e8-8008-361b625d7373.png">

After:
<img width="1399" alt="screen shot 2018-11-11 at 12 46 52 am" src="https://user-images.githubusercontent.com/313208/48310893-5b304c80-e54b-11e8-8e89-5428052614f0.png">

I think this is good to go to beta at least. A couple things were changed that are *not* under a feature flag:

1. Switched the "favorite" tag to a heart instead of a star, so it doesn't overlap with the ratings star.
2. Remove count for stacks of one.

Everything else is only on when the feature flag is on:

1. Taller tiles that show the whole item picture.
2. Element icon next to power.
2. Lock icon for locked items.
3. Tag/lock now in the bottom left, and are green. They're green because it's the only color that'll stand out against yellow, purple, and blue.
4. Ratings are smaller and have a star icon, and only get colored if it's a perfect 5.
5. Most sizing is proportional to the tile size, so adjusting the size (including in mobile) should work better.
5. Use the in-game highlighting for masterworks, but inset it so it doesn't go over our border.
6. Moved the progress bar for objective-laden items to the top, and made it thicker.

<img width="319" alt="screen shot 2018-11-11 at 12 51 49 am" src="https://user-images.githubusercontent.com/313208/48310928-18bb3f80-e54c-11e8-8bfe-34f537f1deae.png">

I tried with and without the ammo indicator, and decided on not, since people can memorize what weapons take what ammo pretty easily.

D1 requires a little polish but isn't bad:

<img width="510" alt="screen shot 2018-11-11 at 12 54 06 am" src="https://user-images.githubusercontent.com/313208/48310956-82d3e480-e54c-11e8-9444-4da49986ae14.png">

There's also sort of a bug where any tile that doesn't have a label is too tall, and the move popup kind of attaches to nowhere. I can fix that up pretty easily later.

bhollis commented:
<img width="589" alt="screen shot 2018-11-11 at 1 03 36 am" src="https://user-images.githubusercontent.com/313208/48311024-aba8a980-e54d-11e8-9328-1a76b24682f8.png">

I'm tempted to space them out a bit more...

bhollis commented:
I'll definitely have to try it - the 1px border looks better now than the last time I tried it, especially if I can use that to get a bit more space between items.

Making reviews small is definitely odd, but rounding is even weirder. There's just still a lot of difference between 4.0 and 5.0. And if I make them full size they but right up against the element. I also really like having the element in the tag.

The lime green icons take some getting used to, but at least they stand out! I was also thinking about having a gradient behind them, and I can try that too.

I don't think having lock override tags is great, because they sorta mean different things. I also tried the experiment of replacing the ratings star with the tag, but it gets confusing then to explain what the number is for.

bhollis commented:
<img width="1355" alt="screen shot 2018-11-11 at 4 51 50 pm" src="https://user-images.githubusercontent.com/313208/48320782-21eeef80-e5d2-11e8-9d28-e502ab32cc84.png">

Updated for thinner borders and more space between tiles, plus some other tweaks.

bhollis commented:
BTW, looking at it, any darker and contrast suffers.

kyleshay commented on src/app/inventory/TallTile.tsx:
```diff
@@ -0,0 +1,117 @@
+  const borderless =
+    (item.isDestiny2 &&
+      item.isDestiny2() &&
+      (item.bucket.hash === 3284755031 || item.itemCategoryHashes.includes(268598612))) ||
```
this line is throwing an error on the settings page (itemCategoryHashes is undefined on our fake item display object.)

duckmanBR commented:
Other than the taller tiles, eagerly waiting for this to be merged on beta...
